diff --git a/kernel/drivers/net/can/kvaser_pciefd/kvaser_pciefd_core.c b/kernel/drivers/net/can/kvaser_pciefd/kvaser_pciefd_core.c
index b4f0da4..ea440a7 100644
--- a/kernel/drivers/net/can/kvaser_pciefd/kvaser_pciefd_core.c
+++ b/kernel/drivers/net/can/kvaser_pciefd/kvaser_pciefd_core.c
@@ -557,6 +557,48 @@ static inline void kvaser_pciefd_abort_flush_reset(struct kvaser_pciefd_can *can
 	kvaser_pciefd_send_kcan_cmd(can, KVASER_PCIEFD_KCAN_CMD_AT);
 }
 
+static ssize_t kv_ean_show(struct device *dev,
+			   struct device_attribute *attr, char *buf)
+{
+	struct kvaser_pciefd_can *can = netdev_priv(to_net_dev(dev));
+	struct kvaser_pciefd *pcie = can->kv_pcie;
+
+	return sprintf(buf, "%x%08x", pcie->ean[1], pcie->ean[0]);
+}
+
+static ssize_t kv_serial_show(struct device *dev,
+			      struct device_attribute *attr, char *buf)
+{
+	struct kvaser_pciefd_can *can = netdev_priv(to_net_dev(dev));
+	struct kvaser_pciefd *pcie = can->kv_pcie;
+
+	return sprintf(buf, "%u\n", pcie->serial_number);
+}
+
+static ssize_t kv_channel_show(struct device *dev,
+			       struct device_attribute *attr, char *buf)
+{
+	struct kvaser_pciefd_can *can = netdev_priv(to_net_dev(dev));
+
+	return sprintf(buf, "%u\n", can->can.dev->dev_id);
+}
+
+static DEVICE_ATTR_RO(kv_ean);
+static DEVICE_ATTR_RO(kv_serial);
+static DEVICE_ATTR_RO(kv_channel);
+
+static struct attribute *kvaser_pciefd_dev_attrs[] = {
+	&dev_attr_kv_ean.attr,
+	&dev_attr_kv_serial.attr,
+	&dev_attr_kv_channel.attr,
+	NULL,
+};
+
+static const struct attribute_group kvaser_pciefd_dev_group = {
+	.attrs = kvaser_pciefd_dev_attrs,
+};
+
+
 static void kvaser_pciefd_enable_err_gen(struct kvaser_pciefd_can *can)
 {
 	u32 mode;
@@ -1090,6 +1132,7 @@ static int kvaser_pciefd_setup_can_ctrls(struct kvaser_pciefd *pcie)
 
 		netdev->flags |= IFF_ECHO;
 		SET_NETDEV_DEV(netdev, &pcie->pci->dev);
+		netdev->sysfs_groups[0] = &kvaser_pciefd_dev_group;
 
 		iowrite32(GENMASK(31, 0), can->reg_base + KVASER_PCIEFD_KCAN_IRQ_REG);
 		iowrite32(KVASER_PCIEFD_KCAN_IRQ_ABD,
diff --git a/kernel/drivers/net/can/usb/kvaser_usb/kvaser_usb_core.c b/kernel/drivers/net/can/usb/kvaser_usb/kvaser_usb_core.c
index 21eda50..8dd192d 100644
--- a/kernel/drivers/net/can/usb/kvaser_usb/kvaser_usb_core.c
+++ b/kernel/drivers/net/can/usb/kvaser_usb/kvaser_usb_core.c
@@ -802,6 +802,47 @@ freeurb:
 	return ret;
 }
 
+static ssize_t kv_ean_show(struct device *dev,
+			   struct device_attribute *attr, char *buf)
+{
+	struct kvaser_usb_net_priv *priv = netdev_priv(to_net_dev(dev));
+	struct kvaser_usb *kvaser_dev = priv->dev;
+
+	return sprintf(buf, "%x%08x", kvaser_dev->ean[1], kvaser_dev->ean[0]);
+}
+
+static ssize_t kv_serial_show(struct device *dev,
+			      struct device_attribute *attr, char *buf)
+{
+	struct kvaser_usb_net_priv *priv = netdev_priv(to_net_dev(dev));
+	struct kvaser_usb *kvaser_dev = priv->dev;
+
+	return sprintf(buf, "%u\n", kvaser_dev->serial_number);
+}
+
+static ssize_t kv_channel_show(struct device *dev,
+			       struct device_attribute *attr, char *buf)
+{
+	struct kvaser_usb_net_priv *priv = netdev_priv(to_net_dev(dev));
+
+	return sprintf(buf, "%u\n", priv->channel);
+}
+
+static DEVICE_ATTR_RO(kv_ean);
+static DEVICE_ATTR_RO(kv_serial);
+static DEVICE_ATTR_RO(kv_channel);
+
+static struct attribute *kvaser_usb_dev_attrs[] = {
+	&dev_attr_kv_ean.attr,
+	&dev_attr_kv_serial.attr,
+	&dev_attr_kv_channel.attr,
+	NULL,
+};
+
+static const struct attribute_group kvaser_usb_dev_group = {
+	.attrs = kvaser_usb_dev_attrs,
+};
+
 static int kvaser_usb_set_phys_id(struct net_device *netdev,
 				  enum ethtool_phys_id_state state)
 {
@@ -954,6 +995,7 @@ static int kvaser_usb_init_one(struct kvaser_usb *dev, int channel)
 
 	netdev->flags |= IFF_ECHO;
 
+	netdev->sysfs_groups[0] = &kvaser_usb_dev_group;
 	netdev->netdev_ops = &kvaser_usb_netdev_ops;
 	netdev->ethtool_ops = &kvaser_usb_ethtool_ops;
 	SET_NETDEV_DEV(netdev, &dev->intf->dev);
